.github/pull_request_template.md

```diff
@@ -0,0 +1,27 @@
+# Description
+
+Please include a summary of the change and which issue is fixed. 
+Please also include relevant motivation and context. 
+List any dependencies that are required for this change.
+
+
+# Checklist:
+
+Here is a checklist that should completed before merging this given feature. 
+Any shortcomings from the items below should be explained and detailed within the contents of this PR.
+
+- [ ] I am proud of of the feature.
+- [ ] I have performed a self-review of my own code.
```

Comment:
This is quite an extensive list. Is every bullet point intended to be mandatory for all (feature) PRs?

Reply:
Yup. I am open to suggestions though. What should not be included?